application/src/main/java/org/opentripplanner/transit/service/TransitService.java

```diff
@@ -169,6 +169,18 @@ public interface TransitService {
    */
   Collection<Trip> getAllTrips();
 
+  /**
+   * Find canceled trips.
+   *
+   * @param feeds used for filtering. Empty list is not allowed.
+   */
+  List<TripOnServiceDate> findCanceledTrips(List<String> feeds);
```

Comment:
It's sometimes easier to group geographical areas (such as cities) in feeds rather than agencies, at least in my project. However, feed is not useful if you have one data set that contains everything. How agencies are defined in data differ a lot in different data sets. Sometimes the agency is something like HSL which doesn't operate the busses themselves but rather just contract bus companies to operate the lines/routes. In other cases, the agency is the operator. I think in theory the correct way to model agencies in GTFS would be to use HSL as the agency rather than the operators, but I'm not even sure. The reason why sometimes the individual bus operators are listed as agencies is probably due to some integrations which requires to differentiate between different operators.

We for example have data sets which have 10+ agencies operating the traffic for one city and the agencies constantly change. To use agencies for the filtering, we would need to fetch the current list of agencies for the feed first and then use the 10+ agencies as filtering criteria.

I don't think we can get rid of the feed in OTP in the near future, if ever. There are a few options:
1. We just have the feeds argument in the query as is and if need some other type of filter criteria in the future, we then figure out what we should do.
2. We now implement a more complex filter API for this purpose (which might only have feed as the only option for now).
3. We force people to fetch the agencies that they are interested in before using this query and then just have agencies as the filtering criteria either directly on the query or though a more complex filter API structure.

Reply:
I don't think that we should filter by feeds. Feeds are an implementation detail and there shouldn't be a difference if the transport data are delivered in one feed or in multiple feeds. In contrast, agency code should be relatively stable and easy to fetch.

For example, previously my project had mainline rail and other transport in two different feeds, but now I have combined them into one feed with some data improvement. None of the agency IDs have changed as they are all national standard code.

Also, in Transmodel, there is a distinction between Authority and Operator, while GTFS does not make this distinction. In GTFS, the agency should be the entity seen from the passenger's perspective, i.e. HSL.

Reply:
"None of the agency IDs have changed as they are all national standard code" something like this rarely exists in GTFS data sets. Usually the ids are quite random -- they can change daily and the same agency can have different ids in different feeds (although don't know how much this matters since we probably need to use a feed scoped vearsion of agency id for filtering anyway). You can check [here](https://api.digitransit.fi/graphiql/finland/v2?query=%257B%250A%2520%2520feeds%2520%257B%250A%2520%2520%2520%2520agencies%2520%257B%250A%2520%2520%2520%2520%2520%2520name%250A%2520%2520%2520%2520%2520%2520gtfsId%250A%2520%2520%2520%2520%257D%250A%2520%2520%257D%250A%257D) what kind of a mess the agencies are in Finland, for example.